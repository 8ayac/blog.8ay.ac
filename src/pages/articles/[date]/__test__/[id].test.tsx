import ArticlePage from '@/src/pages/articles/[date]/[id]';
import { mount } from 'enzyme';
import React from 'react';

describe('ArticlePage', () => {
  const testArticleData = {
    id: 'example-id',
    title: 'Example01',
    publishedAt: '2021-01-02 03:04:56 +0900',
    updatedAt: '2021-01-02 03:04:56 +0900',
    body: '## test\r\n\r\nbluhbluhbluh',
  };

  it('is rendered correctly to match the snapshot', () => {
    const wrapper = mount(<ArticlePage article={testArticleData} />);
    expect(wrapper).toMatchSnapshot();
  });
});
