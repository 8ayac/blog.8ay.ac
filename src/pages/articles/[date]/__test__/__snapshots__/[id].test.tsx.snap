// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticlePage is rendered correctly to match the snapshot 1`] = `
<Fragment>
  <ArticleWrapperDiv>
    <ArticleHeader
      article={
        Object {
          "body": "## test

bluhbluhbluh",
          "id": "example-id",
          "publishedAt": "2021-01-02 03:04:56 +0900",
          "title": "Example01",
          "updatedAt": "2021-01-02 03:04:56 +0900",
        }
      }
    />
    <MarkdownRender
      content="## test

bluhbluhbluh"
    />
  </ArticleWrapperDiv>
</Fragment>
`;

exports[`Styled component used in ArticlePage to match the snapshot S.ArticleWrapperDiv 1`] = `
.emotion-0 {
  font-size: 1.6rem;
  word-break: break-all;
}

<div
  className="emotion-0 emotion-1"
/>
`;
